### PR TITLE
Manage access permission test

### PR DESCRIPTION
**Pull request by mmusil**
Until refactored access dialog is released, this test needs to run as org admin and requires a secondary user to test if the permission was really applied and works as expected.

The prerequisite for this is this PR to mas-ci: https://gitlab.cee.redhat.com/mk-ci-cd/mas-ci/-/merge_requests/677

### CODE REVIEW

**Comment by mmusil** [comment]
Will add some condition to skip steps that need a second user if credentials are not supplied tomorrow.

File: tests/access.spec.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      if (name !== testInstanceName) {
+        await deleteKafkaInstance(page, name);
+      }
+    }
+
+    if ((await page.getByText(testInstanceName).count()) === 0) {
+      await createKafkaInstance(page, testInstanceName);
+    }
+  }
+});
```
Comment (Frawless) [inline tests/access.spec.ts:43]: This is something that could be generalized. We should think about it in future PRs.

File: tests/access.spec.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  await login(page, config.username_secondary, config.password_secondary);
+
+  await navigateToAccess(page, testInstanceName);
+  await grantManageAccess(page, 'All accounts');
+
+  const rowAllAccounts = await findAccessRow(page, 'All Accounts', 'Alter', 'Kafka Instance');
+  await expect(rowAllAccounts).toHaveCount(1);
+
+  await revokeAccess(page, 'All Accounts', 'Alter', 'Kafka Instance');
+  await revokeAccess(page, config.username_secondary, 'Alter', 'Kafka Instance');
```
Comment (Frawless) [inline tests/access.spec.ts:63]: Shouldn't be there some kind of assert?

File: tests/access.spec.ts (hunk 1 above)
Comment (mmusil) [reply tests/access.spec.ts:43]: Absolutely valid point and agree.

File: lib/auth.ts [hunk 3]
```diff
@@ -27,6 +27,9 @@ export default async function login(
 }
 
 export async function logout(page: Page) {
-  await page.getByRole('button', { name: /User Avatar/ }).click();
-  await page.getByRole('menuitem', { name: 'Log out' }).click();
+  // await page.getByRole('button', { name: /User Avatar/ }).click();
+  // await page.getByRole('menuitem', { name: 'Log out' }).click();
```
[the remark this answers is not in the document]
Comment (mmusil) [reply lib/auth.ts:31]: I wanted to ask if anybody would have a few minutes to test this on their laptop and try either variant. To know if it is really just my machine that does not like that getByRole for some reason.

File: tests/access.spec.ts (hunk 2 above)
Comment (mmusil) [reply tests/access.spec.ts:63]: I was thinking about this more like a test teardown cleanup, but yeah, will add something.

File: lib/config.ts [hunk 4]
```diff
@@ -3,6 +3,8 @@ import { v1 as uuid } from 'uuid';
 class Config {
   readonly username: string;
   readonly password: string;
+  readonly username_secondary: string;
+  readonly password_secondary: string;
```
[the remark this answers is not in the document]
Comment (mmusil) [reply lib/config.ts]: Changed.

File: tests/access.spec.ts (hunk 2 above)
Comment (mmusil) [reply tests/access.spec.ts]: Added assert directly to the function.

**Review by Frawless — APPROVED** [review]

**Review by kornys — APPROVED** [review]

**Comment by mmusil** [comment]
Forgot to wait for the instance to be ready 🤦‍♂️. Next build should pass 😀

**Review by kornys — APPROVED** [review]

**Pull request merged**